application/client.core/src/app/environment/services/service.file.opener.ts

```diff
@@ -68,50 +68,61 @@ export class FileOpenerService implements IService, IFileOpenerService {
         });
     }
 
-    public open(files: IFile[]) {
-        if (files.length === 0) {
-            return;
-        }
-        if (files.length === 0) {
-            return;
-        }
-        this._setSessionForFile().then((session: ControllerSessionTab) => {
-            TabsSessionsService.setActive(session.getGuid());
-            if (files.length === 1) {
-                // Single file
-                ServiceElectronIpc.request(new IPCMessages.FileOpenRequest({
-                    file: files[0].path,
-                    session: session.getGuid(),
-                }), IPCMessages.FileOpenResponse).then((response: IPCMessages.FileReadResponse) => {
-                    if (response.error !== undefined) {
-                        this._logger.error(`Fail open file "${files[0].path}" due error: ${response.error}`);
-                        // TODO: add notification here
-                        return;
+    public open(files: IFile[]): Promise<Error> {
+        return new Promise((resolve, reject) => {
+            if (files.length === 0) {
+                return;
+            }
+            ServiceElectronIpc.request(new IPCMessages.FileListRequest({
+                files: files.map((file: IFile) => file.path),
+            }), IPCMessages.FileListResponse).then((checkResponse: IPCMessages.FileListResponse) => {
+                this._setSessionForFile().then((session: ControllerSessionTab) => {
+                    TabsSessionsService.setActive(session.getGuid());
+                    if (checkResponse.error !== undefined) {
+                        const errorMessage = `Fail check paths due error: ${checkResponse.error}`;
+                        this._logger.error(errorMessage);
+                        reject(new Error(errorMessage));
+                    }
+                    if (checkResponse.files.length === 1) {
```

Comment:
function will be rejected, but not stopped. JS will go to line 86 after `reject`